CMakeLists.txt

```diff
@@ -104,114 +49,19 @@ if(WHOLE_PROGRAM_OPTIMISATION)
 	set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE ${IPO_SUPPORTED})
 endif()
 
-# Static CRT
+# Static CRT:
 set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
 
-# This has to be done before any flags have been set up.
-if(${BUILD_TOOLS})
-	message("Building tools")
-	add_subdirectory(Tools/GrownBiomeGenVisualiser/)
-	add_subdirectory(Tools/MCADefrag/)
-	add_subdirectory(Tools/NoiseSpeedTest/)
-	add_subdirectory(Tools/ProtoProxy/)
-endif()
-
-if(${BUILD_UNSTABLE_TOOLS})
-	message("Building unstable tools")
-	add_subdirectory(Tools/GeneratorPerformanceTest/)
-endif()
-
-include(SetFlags.cmake)
-set_flags()
-set_lib_flags()
-enable_profile()
-
-# Set options for SQLiteCpp, disable all their tests and lints:
-set(SQLITECPP_RUN_CPPLINT     OFF CACHE BOOL "Run cpplint.py tool for Google C++ StyleGuide."  FORCE)
-set(SQLITECPP_RUN_CPPCHECK    OFF CACHE BOOL "Run cppcheck C++ static analysis tool."          FORCE)
-set(SQLITECPP_RUN_DOXYGEN     OFF CACHE BOOL "Run Doxygen C++ documentation tool."             FORCE)
-set(SQLITECPP_BUILD_EXAMPLES  OFF CACHE BOOL "Build examples."                                 FORCE)
-set(SQLITECPP_BUILD_TESTS     OFF CACHE BOOL "Build and run tests."                            FORCE)
-set(SQLITECPP_INTERNAL_SQLITE ON  CACHE BOOL "Add the internal SQLite3 source to the project." FORCE)
-set(SQLITE_ENABLE_COLUMN_METADATA OFF CACHE BOOL "" FORCE)
-
-# Set options for LibEvent, disable all their tests and benchmarks:
-set(EVENT__DISABLE_OPENSSL   YES CACHE BOOL   "Disable OpenSSL in LibEvent"       FORCE)
-set(EVENT__DISABLE_BENCHMARK YES CACHE BOOL   "Disable LibEvent benchmarks"       FORCE)
-set(EVENT__DISABLE_TESTS     YES CACHE BOOL   "Disable LibEvent tests"            FORCE)
-set(EVENT__DISABLE_REGRESS   YES CACHE BOOL   "Disable LibEvent regression tests" FORCE)
-set(EVENT__DISABLE_SAMPLES   YES CACHE BOOL   "Disable LibEvent samples"          FORCE)
-set(EVENT__LIBRARY_TYPE "STATIC" CACHE STRING "Use static LibEvent libraries"     FORCE)
+# TODO: set_build_stamp()
```

Comment:
Something unfinished?

Reply:
I'm planning to, but I decided against making this diff any larger, you'd kill me

Reply:
If this is missing functionality then can't you just leave the original code?

Reply:
Sorry sorry, StampBuild.cmake is included and adds the build id as before.

But it just executes the file right now. The todo is for making it a function as you suggested for the others. Then we can call set_build_stamp(Cuberite)

Reply:
I see. Alright then.